src/hooks/plan_tree_walker.c

```diff
@@ -0,0 +1,106 @@
+#include <postgres.h>
+
+#include <miscadmin.h>
+#include <nodes/nodeFuncs.h>
+#include <nodes/parsenodes.h>
+#include <nodes/plannodes.h>
+
+bool plan_tree_walker_util(Plan *plan, bool (*walker_func)(Plan *plan, void *context), void *context)
```

Comment:
This function is like `any` over a collection type that checks the predicate (walker_func) in this case
I'd name this
`bool plan_tree_walker_any(Plan *plan, bool (*predicate_func)(Plan *plan, void *context), void *context)`

other names work too, but would be good if the name communicates when the function returns true


`

Reply:
`plan_tree_walker` is for plan nodes. `plan_tree_walker_util` is for `Plan`, which is a specific type of Plan node that is shared across all plan nodes. The naming isn't the best, I agree. Will adjust.